File: apps/api/app/schemas/startup_discovery.py
from decimal import Decimal
from uuid import UUID

from pydantic import BaseModel, Field


class StartupDiscoveryItem(BaseModel):
    id: UUID
    user_id: UUID
    startup_name: str
    industry: str
    website_url: str | None = None
    headquarters: str | None = None
    founded_year: int | None = None
    stage: str | None = None
    business_model: str | None = None
    target_market: str | None = None
    description: str
    funding_required: Decimal
    monthly_revenue: Decimal | None = None
    annual_recurring_revenue: Decimal | None = None
    gross_margin_percent: Decimal | None = None
    net_profit: Decimal | None = None
    burn_rate: Decimal | None = None
    runway_months: int | None = None
    customer_count: int | None = None
    valuation: Decimal | None = None
    revenue_projection_year1: Decimal | None = None
    revenue_projection_year2: Decimal | None = None
    revenue_projection_year3: Decimal | None = None
    profit_projection_year1: Decimal | None = None
    profit_projection_year2: Decimal | None = None
    profit_projection_year3: Decimal | None = None
    patents_filed: int | None = None
    patents_granted: int | None = None
    traction_summary: str | None = None
    use_of_funds: str | None = None
    pitch_video_url: str | None = None
    verification_badges: list[str] = Field(default_factory=list)

    model_config = {"from_attributes": True}


class StartupDiscoveryResponse(BaseModel):
    items: list[StartupDiscoveryItem]
    total: int
    limit: int
    offset: int

File: apps/web/src/lib/api/startup-discovery.ts
import { apiClient } from "@/lib/api/client";
import type { VerificationBadge } from "@/lib/api/profile-verification";

export type StartupDiscoveryItem = {
  id: string;
  user_id: string;
  startup_name: string;
  industry: string;
  website_url: string | null;
  headquarters: string | null;
  founded_year: number | null;
  stage: string | null;
  business_model: string | null;
  target_market: string | null;
  description: string;
  funding_required: string;
  monthly_revenue: string | null;
  annual_recurring_revenue: string | null;
  gross_margin_percent: string | null;
  net_profit: string | null;
  burn_rate: string | null;
  runway_months: number | null;
  customer_count: number | null;
  valuation: string | null;
  revenue_projection_year1: string | null;
  revenue_projection_year2: string | null;
  revenue_projection_year3: string | null;
  profit_projection_year1: string | null;
  profit_projection_year2: string | null;
  profit_projection_year3: string | null;
  patents_filed: number | null;
  patents_granted: number | null;
  traction_summary: string | null;
  use_of_funds: string | null;
  pitch_video_url: string | null;
  verification_badges: VerificationBadge[];
};

export type StartupDiscoveryResponse = {
  items: StartupDiscoveryItem[];
  total: number;
  limit: number;
  offset: number;
};

export type StartupDiscoveryParams = {
  query?: string;
  industry?: string;
  funding_min?: string;
  funding_max?: string;
  limit?: number;
  offset?: number;
};

export async function discoverStartups(params: StartupDiscoveryParams) {
  const response = await apiClient.get<StartupDiscoveryResponse>("/startup-discovery", {
    params
  });

  return response.data;
}
